### DISCUSSION TITLE
dll requirment

### DISCUSSION THREAD

**Discussion by itsabhist** [iterativ/openopc2#59] [General]
I'm currently exploring the OpenOPC2 package for Python to interface with OPC servers in my power plant. While going through the documentation, I came across mentions of the graybox DLL required for registering and connecting to OPC DA servers. However, I noticed that I was able to fetch data from my OPC server using a test script without explicitly using the DLL. Here's the test script I used: [test_read.py](https://github.com/iterativ/openopc2/blob/develop/tests%2Ftest_read.py). Could you please clarify when the graybox DLL is required and how it relates to accessing OPC servers using OpenOPC2?

**Comment by renzop** [comment]
Hi, 

The graybox automation wrapper (dll) is a wrapper around your OPC server. Even if you don't register it you could work with openopc2. (I never tested it but it makes sense). I guess some error messages could be probably less understandable that with the dll. I might be that you already have registered a dll with a similar functionality. (http://gray-box.net/daawrapper.php?lang=en) 

It is one part of the project that we took over from the original OpenOPC, and we did not touch it. I am really not an expert in that area.

**Reply by itsabhist** [reply]
![IMG_20240228_121131.jpg](https://github.com/iterativ/openopc2/assets/36887926/5b6901af-4c55-406d-a863-e93e9448d631)

**Reply by itsabhist** [reply]
I am trying to connect my office server using exe file

I have set environment variable as shown in list-config

But but while I am trying to get list-server
It's showing error

As shown in above image

**Reply by renzop** [reply]
Please do not post screen photos. Please plost the terminal output as text.

**Reply by itsabhist** [reply]
Ok

**Comment by itsabhist** [comment]
`(c) 2018 Microsoft Corporation. All rights reserved.

C:\Users\Administrator\Desktop\OPC>OpenOpcCli.exe --help

 Usage: OpenOpcCli.exe [OPTIONS] COMMAND [ARGS]...

+- Options -----------------------------------------------------------------------------------------------------------+
¦ --install-completion          Install completion for the current shell.                                             ¦
¦ --show-completion             Show completion for the current shell, to copy it or customize the installation.      ¦
¦ --help                        Show this message and exit.                                                           ¦
+---------------------------------------------------------------------------------------------------------------------+
+- Commands ----------------------------------------------------------------------------------------------------------+
¦ list-clients            [EXPERIMENTAL] List clients of OpenOPC Gateway Server                                       ¦
¦ list-config             Write values                                                                                ¦
¦ list-servers            List all OPC DA servers                                                                     ¦
¦ list-tags               List tags (items) of OPC server                                                             ¦
¦ properties              Show properties of given tags                                                               ¦
¦ read                    Read tags                                                                                   ¦
¦ server-info             Display OPC server information                                                              ¦
¦ write                   Write values                                                                                ¦
+---------------------------------------------------------------------------------------------------------------------+


C:\Users\Administrator\Desktop\OPC>OpenOpcCli.exe server-info
[12:08:29] ERROR    Could not connect to OPC server: unsupported serialized class: pywintypes.com_error      cli.py:344

C:\Users\Administrator\Desktop\OPC>OpenOpcCli.exe list-config
Open Opc Config:
OPC_HOST              : localhost
OPC_SERVER            : Ovation.OPCDAServer.1
OPC_CLIENT            : OpenOPC
OPC_GATEWAY_HOST      : 192.168.2.180
OPC_GATEWAY_PORT      : 7766
OPC_CLASS             : Graybox.OPC.DAWrapper
OPC_MODE              : com
OPC_TIMEOUT           : 1000

C:\Users\Administrator\Desktop\OPC>OpenOpcCli.exe list-servers
[12:12:08] ERROR    Could not connect to OPC server: unsupported serialized class: pywintypes.com_error                                                                                                                           cli.py:344

C:\Users\Administrator\Desktop\OPC>OpenOpcCli.exe list-servers
[09:07:15] ERROR    Could not connect to OPC server: unsupported serialized class: pywintypes.com_error                                                                                                                           cli.py:344

C:\Users\Administrator\Desktop\OPC>OpenOpcCli.exe list-servers`


here is error
